Makefile

```diff
@@ -1,4 +1,17 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+ifndef BUILD_DIRECTORY
+	$(error You should define a build directory (set BUILD_DIRECTORY))
+endif
+
+BUILD_TEMPORARY_DIRECTORY = $(BUILD_DIRECTORY)/.tmp
+BUILD_OUT_DIRECTORY = $(BUILD_DIRECTORY)/out
```

Comment:
IMHO `BUILD_OUTPUT_DIRECTORY` or even `BUILD_INSTALLER_DIRECTORY` might be better names here?

To me, BUILD_OUT_DIRECTORY sounds like 'buildout directory'.